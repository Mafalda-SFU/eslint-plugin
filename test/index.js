const {expect, use} = require('chai')
const {test} = require('mocha')
const {jestSnapshotPlugin} = require('mocha-chai-jest-snapshot')

const plugin = require('../lib/index.js')


use(jestSnapshotPlugin())


test('layout', function()
{
  expect(plugin).toMatchSnapshot()
})
