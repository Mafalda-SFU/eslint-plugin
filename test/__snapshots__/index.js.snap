// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout 1`] = `
Object {
  "configs": Object {
    "layout": Object {
      "plugins": Array [
        "@mafalda-sfu",
      ],
      "rules": Object {
        "@mafalda-sfu/brace-style": Array [
          "error",
          "allman-multiline",
        ],
        "brace-style": "off",
      },
    },
    "recommended": Object {
      "plugins": Array [
        "@mafalda-sfu",
      ],
      "rules": Object {
        "@mafalda-sfu/brace-style": Array [
          "error",
          "allman-multiline",
        ],
        "@mafalda-sfu/no-setInterval": "error",
        "@mafalda-sfu/no-setTimeout": "warn",
        "@mafalda-sfu/no-window-event": "error",
        "brace-style": "off",
      },
    },
    "suggestions": Object {
      "plugins": Array [
        "@mafalda-sfu",
      ],
      "rules": Object {
        "@mafalda-sfu/no-setInterval": "error",
        "@mafalda-sfu/no-setTimeout": "warn",
        "@mafalda-sfu/no-window-event": "error",
      },
    },
    "suggestions-typescript": Object {
      "plugins": Array [
        "@mafalda-sfu",
      ],
      "rules": Object {
        "@mafalda-sfu/no-private-keyword": "error",
        "@mafalda-sfu/no-protected-keyword": "warn",
        "@mafalda-sfu/no-public-keyword": "error",
      },
    },
    "typescript": Object {
      "plugins": Array [
        "@mafalda-sfu",
      ],
      "rules": Object {
        "@mafalda-sfu/no-private-keyword": "error",
        "@mafalda-sfu/no-protected-keyword": "warn",
        "@mafalda-sfu/no-public-keyword": "error",
      },
    },
  },
  "rules": Object {
    "brace-style": Object {
      "create": [Function],
      "meta": Object {
        "docs": Object {
          "description": "Enforce consistent brace style for blocks",
          "recommended": false,
          "url": "https://eslint.org/docs/latest/rules/brace-style",
        },
        "fixable": "whitespace",
        "messages": Object {
          "blockSameLine": "Statement inside of curly braces should be on next line.",
          "nextLineClose": "Closing curly brace does not appear on the same line as the subsequent block.",
          "nextLineOpen": "Opening curly brace does not appear on the same line as controlling statement.",
          "sameLineClose": "Closing curly brace appears on the same line as the subsequent block.",
          "sameLineOpen": "Opening curly brace appears on the same line as controlling statement.",
          "singleLineClose": "Closing curly brace should be on the same line as opening curly brace or on the line after the previous block.",
        },
        "schema": Array [
          Object {
            "enum": Array [
              "1tbs",
              "stroustrup",
              "allman",
              "allman-multiline",
            ],
          },
          Object {
            "additionalProperties": false,
            "properties": Object {
              "allowSingleLine": Object {
                "default": false,
                "type": "boolean",
              },
            },
            "type": "object",
          },
        ],
        "type": "layout",
      },
    },
    "no-private-keyword": [Function],
    "no-protected-keyword": [Function],
    "no-public-keyword": [Function],
    "no-setInterval": Object {
      "create": [Function],
      "meta": Object {
        "fixable": "code",
        "messages": Object {},
      },
    },
    "no-setTimeout": Object {
      "create": [Function],
      "meta": Object {
        "fixable": "code",
        "messages": Object {},
      },
    },
    "no-window-event": Object {
      "create": [Function],
      "meta": Object {
        "fixable": "code",
        "messages": Object {},
      },
    },
  },
}
`;
